docs/proposals/commit-push.md

```diff
@@ -0,0 +1,158 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# commit and push capability proposal
+
+### Motivation
+
+There are some use case to commit a container, and push its resulting image to a registry, for example:
+
+1. For a run-once pod whose result represents an image that should be pushed to a registry for consumption
+by other actors in the system
+
+2. There is someone desire to be able to modify a running container and then commit/push the change so they
+could, for example, make a debug change and then scale up the deployment to test it in a clustered mode,
+without having to start from scratch with building a new image.
+
+The goal of this proposal is to add the commit/push capability, which will allow user commit+push their
+containers manually, or automically after container exits.
+
+
+
+### Design
+
+1. Add `pod/commit` and `pod/push` subsource to support user commit+push their containers manually.
+
+2. Add `PosStop` life-cycle hook to support commit+push automically after container exits.
+
+
+### API
+
+1) We will add
+
+```
+ImagePushSecrets []LocalObjectReference `json:"imagePushSecrets,omitempty"`
```

Comment:
I don't think this belongs to the pod, because pushing images is not part of the declarative spec.  I would expect a distinct resource to represent that push - for instance for this to be PodCommitOptions